graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphqlServiceBuilder.java

```diff
@@ -65,32 +67,57 @@ public final class GraphqlServiceBuilder {
 
     private static final List<String> DEFAULT_SCHEMA_FILE_NAMES = ImmutableList.of("schema.graphqls",
                                                                                    "schema.graphql");
-    private final ImmutableList.Builder<URL> schemaUrls = ImmutableList.builder();
-
-    private final ImmutableList.Builder<RuntimeWiringConfigurator> runtimeWiringConfigurators =
-            ImmutableList.builder();
-    private final ImmutableList.Builder<GraphQLTypeVisitor> typeVisitors = ImmutableList.builder();
-    private final ImmutableList.Builder<Instrumentation> instrumentations = ImmutableList.builder();
-    private final ImmutableList.Builder<GraphqlConfigurator> graphqlBuilderConsumers =
-            ImmutableList.builder();
 
     @Nullable
-    private ImmutableList.Builder<Consumer<? super DataLoaderRegistry>> dataLoaderRegistryConsumers;
-    private boolean useBlockingTaskExecutor;
+    private GraphQL graphql;
 
+    // Fields for building a graphql
     @Nullable
-    private Function<? super ServiceRequestContext, ? extends DataLoaderRegistry> dataLoaderRegistryFactory;
+    private ExecutionIdGenerator executionIdGenerator;
+    @Nullable
+    private ImmutableList.Builder<Instrumentation> instrumentations;
+    @Nullable
+    private ImmutableList.Builder<GraphqlConfigurator> graphqlBuilderConsumers;
 
+    // Fields for building a schema in a graphql
     @Nullable
     private GraphQLSchema schema;
+    @Nullable
+    private ImmutableList.Builder<URL> schemaUrls;
+    @Nullable
+    private ImmutableList.Builder<RuntimeWiringConfigurator> runtimeWiringConfigurators;
+    @Nullable
+    private ImmutableList.Builder<GraphQLTypeVisitor> typeVisitors;
 
+    // Fields for building a DataLoaderRegistry
     @Nullable
-    private GraphqlErrorHandler errorHandler;
+    private ImmutableList.Builder<Consumer<? super DataLoaderRegistry>> dataLoaderRegistryConsumers;
+    @Nullable
+    private Function<? super ServiceRequestContext, ? extends DataLoaderRegistry> dataLoaderRegistryFactory;
 
-    private ExecutionIdGenerator executionIdGenerator = ExecutionIdGenerator.of();
+    // Others
+    private boolean useBlockingTaskExecutor;
+    @Nullable
+    private GraphqlErrorHandler errorHandler;
```

Comment:
The two properties are only provided by Armeria and the majority are necessary to build `GraphQL`. Too many `checkState()` looks messy. How about moving properties used to build `GraphQL` to `GraphqlBuilder`?
```java
GraphqlService
  .builder()
  .graphql() // Return GraphqlBuilder
    .schema(...)
    .dataLoaderRegistry(...)
    .and() // Return GraphqlServiceBuilder
  .useBlockingTaskExecutor(true);
  .bulid();


GraphqlService
  .builder()
  .graphql(graphql) // Directly inject GraphQL
  .useBlockingTaskExecutor(true);
  .bulid()
```

Reply:
I was actually thinking about it but `GraphQL` provides its own builder via `new GraphQL.Builder()`.
Also, I found out that our builder which is `GraphqlServiceBuilder` does not provide all setters that the `GraphQL` provides. I think there's always a mismatch unless we keep up the upstream change.
So I was rather deprecating the setters:
https://github.com/line/armeria/pull/5269/files#diff-6b64ad742ab0aa687e003b83510d32495d933e5b3fd2a2236184d6f01bbfcc50R109-R110

> Too many checkState() looks messy.

Yes, but we need it anyway because we can't just remove the setter. 😉

Reply:
> So I was rather deprecating the setters:

I agree with this idea.

Reply:
I've deprecated all the setters. PTAL. 🙇